internal/infrastructure/bot/operations.go

```diff
@@ -461,3 +463,10 @@ func (b *Bot) OpenDM(m *member.Member) (*discord.Channel, error) {
 
 	return MapChannel(channel), nil
 }
+
+func (b *Bot) SetGuildWorld(guildID, world string) error {
+	if b.worldNameRepo == nil {
+		return fmt.Errorf("worldNameRepo is not configured")
+	}
+	return b.worldNameRepo.UpsertGuildWorld(context.Background(), guildID, world)
+}
```

Comment:
Infra layer of Discord should have no knowledge on _how_ do we connect guilds and worlds; move this to `onlinecheck` or any other "core logic" service.